src/main/java/com/fasterxml/jackson/databind/jsontype/impl/SimpleClassNameIdResolver.java

```diff
@@ -0,0 +1,194 @@
+package com.fasterxml.jackson.databind.jsontype.impl;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.BeanDescription;
+import com.fasterxml.jackson.databind.DatabindContext;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.cfg.MapperConfig;
+import com.fasterxml.jackson.databind.jsontype.NamedType;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeSet;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * {@link com.fasterxml.jackson.databind.jsontype.TypeIdResolver} implementation
+ * that converts between (JSON) Strings and simple Java class names via {@link Class#getSimpleName()}.
+ * <p>
+ * Note that this implementation is identical to {@link TypeNameIdResolver} except that instead of 
+ * {@link TypeNameIdResolver#_defaultTypeId(Class)}, this implementation uses {@link Class#getSimpleName()}.
+ * 
+ * @since 2.16
+ */
+public class SimpleClassNameIdResolver
```

Comment:
I think we may want to use `SimpleNameIdResolver` instead of this -- just because `ClassNameIdResolver` has very different behavior: it uses class name to resolve directly into classes (and same with "MinimalClassIdResolver"), whereas here we use logical type name that just happens to default to `Class.getSimpleName()`.